pkg/infra/tracing/tracing.go

```diff
@@ -305,28 +310,35 @@ func (ots *TracingService) initOpentelemetryTracer() error {
 
 	return nil
 }
-
-func (ots *TracingService) Run(ctx context.Context) error {
+func (ots *TracingService) start(ctx context.Context) error {
 	otel.SetErrorHandler(otel.ErrorHandlerFunc(func(err error) {
 		err = level.Error(ots.log).Log("msg", "OpenTelemetry handler returned an error", "err", err)
 		if err != nil {
 			ots.log.Error("OpenTelemetry log returning error", err)
 		}
 	}))
+	return nil
+}
+
+func (ots *TracingService) Run(ctx context.Context) error {
+	if err := ots.StartAsync(ctx); err != nil {
+		return err
+	}
+	return ots.AwaitTerminated(ctx)
+}
+
+func (ots *TracingService) run(ctx context.Context) error {
```

Comment:
nit: I would call this `running`, to avoid having both `Run` (for grafana module) and `run` (for dskit service) methods.

Reply:
updated